.github/workflows/tests.yaml

```diff
@@ -0,0 +1,129 @@
+name: tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  unit-test:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # ratchet:actions/setup-go@v4
+        with:
+          go-version: '1.21'
+
+      - name: Install Dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y clang
+          sudo apt install -y libbpf-dev
+          sudo apt install -y libseccomp-dev
+
+      - name: Build coverage-instrumented binary
+        run: |
+          make build-static-libbpfgo
+          make build-bpf
+
+      - name: Run Unit-Test
+        run: |
+          mkdir /tmp/unit/
+          # test packages excluding the ones with libbpfgo
+          go test \
+            -cover \
+            -v \
+            $(go list ./... | grep -v "github.com/alegrey91/harpoon$" | grep -v ebpf | grep -v cmd) \
+            -skip TestHarpoon \
+            -args -test.gocoverdir=/tmp/unit/
+
+      - name: Upload cover profiles
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # ratchet:actions/upload-artifact@v3
+        with:
+          name: unit-test
+          path: /tmp/unit/
+
+  integration-test:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # ratchet:actions/setup-go@v4
+        with:
+          go-version: '1.21'
+
+      - name: Install Dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y clang
+          sudo apt install -y libbpf-dev
+          sudo apt install -y libseccomp-dev
+
+      - name: Build coverage-instrumented binary
+        run: |
+          make build-static-libbpfgo
+          make build-bpf
+          make build-go-cover && sudo make -B install
+
+      - name: Run integration test
+        run: |
+          mkdir -p /tmp/integration
+          # we have to run integration tests one-by-one
+          # otherwhise they will run in parallel.
+          # since harpoon apply network forwards, these could 
+          # interact with each other and make the test fail.
+          go test \
+            -exec sudo \
+            -cover \
+            -v main_test.go \
+            -args -test.gocoverdir=/tmp/integration/
+
+      - name: Upload cover profiles
+        uses: actions/upload-artifact@v3
+        with:
+          name: integration-test
+          path: /tmp/integration/
+
+  code-coverage:
+
+    runs-on: ubuntu-latest
+    needs: [unit-test,integration-test]
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
No ratchet pining here?

I'm curious why done have, while others don't

Reply:
Forgot about it. I'll add asap.